app_feup/lib/controller/local_storage/app_restaurant_database.dart

```diff
@@ -0,0 +1,117 @@
+
+import 'app_database.dart';
+
+class RestaurantDatabase extends AppDatabase {
+  RestaurantDatabase()
+      : super('restaurant.db',
+      ['CREATE TABLE RESTAURANTS(id INTEGER PRIMARY KEY, ref TEXT , name TEXT)',
+      '''CREATE TABLE MEALS(
+          id INTEGER PRIMARY KEY AUTOINCREMENT,
+          day TEXT,
+          type TEXT,
+          date TEXT,
+          name TEXT,
+          id_restaurant INTEGER,
+          FOREIGN KEY (id_restaurant) REFERENCES RESTAURANTS(id))''']);
+
+  /**
+   * Delets all data, and saves the new restaurants
+   */
+  void saveRestaurants(List<Restaurant> restaurants) async{
+    final Database db = await this.getDatabase();
+    //final Batch batch = db.batch();
+    db.transaction((transaction) async {
+      deleteAll(transaction);
+      restaurants.forEach((restaurant) {
+        insertRestaurant(transaction, restaurant);
+      });
+    });
+
+
+  }
+
+  /**
+   * Get all restaurants and meals, if day is null, all meals are returned
+   */
+  Future<List<Restaurant>> restaurants({DayOfWeek day = null}) async {
+    final Database db = await this.getDatabase();
+    final List<Map<String, dynamic>> restaurantMaps =
+      await db.query('restaurants');
+
+    final List<Restaurant> restaurants =
+      await Future.wait(restaurantMaps.map((map) async {
+        final int restaurantId =  map['id'];
+
+        final List<Meal> meals =
+          await getRestaurantMeals(db, restaurantId, day: day);
+
+        return Restaurant(restaurantId, map['name'], map['ref'], meals: meals);
+    }).toList());
+
+    return restaurants;
+  }
+
+  Future<List<Meal>> getRestaurantMeals(Database db,
+                                        int restaurantId,
+                                        {DayOfWeek day = null}) async{
+    final List<dynamic> whereArgs = [restaurantId];
+    String whereQuery = 'id_restaurant = ? ';
+    if(day != null){
+      whereQuery += ' and day = ?';
+      whereArgs.add(toString(day));
+    }
+
+    //Get restaurant meals
+    final List<Map<String, dynamic>> mealsMaps =
+        await db.query('meals',
+        where: whereQuery,
+        whereArgs: whereArgs);
+
+    //Retreive data from query
+    final List<Meal> meals = mealsMaps.map((map) {
+      final DayOfWeek day = parseDayOfWeek(map['day']);
+      final String type = map['type'];
+      final String name = map['name'];
+      final DateFormat format = DateFormat('d-M-y');
+      final DateTime date = map['date']!= null ?
+      format.parse(map['date']) : null;
+      return Meal(name, type, day, date);
+    }).toList();
+
+    return meals;
+  }
+  /**
+   * Insert restaurant and meals in database
+   */
+  Future<void> insertRestaurant(Transaction txn, Restaurant restaurant) async{
+    final int id = await txn.insert('RESTAURANTS', restaurant.toMap());
+    final Map<String, dynamic> map = (await
+          (txn.query('RESTAURANTS', where:' id = ?', whereArgs: [id]))).first;
```

Comment:
This line is very confusing. Can we indent it to be less confusing? Also, introduce a second variable so we don't have the parenthesis around the `await`, that way it will also make it much more readable.